core/rootvalue.go

```diff
@@ -270,17 +254,34 @@ func (root *RootValue) GetTable(ctx context.Context, tName doltdb.TableName) (*d
 
 // GetTableHash implements the interface doltdb.RootValue.
 func (root *RootValue) GetTableHash(ctx context.Context, tName doltdb.TableName) (hash.Hash, bool, error) {
-	tableMap, err := root.getTableMap(ctx, tName.Schema)
+	_, rawID, objID, err := rootobject.ResolveName(ctx, root, tName)
```

Comment:
You have this logic backwards: you almost certainly want to find this in the table map first, then look in other root object storage only if it's not there.